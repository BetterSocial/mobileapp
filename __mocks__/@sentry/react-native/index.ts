export const addBreadcrumb = jest.fn();
