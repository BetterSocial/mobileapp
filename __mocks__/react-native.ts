import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN