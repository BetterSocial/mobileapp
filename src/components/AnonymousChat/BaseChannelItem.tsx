import * as React from 'react';
import {View} from 'react-native';

import ChannelImage from './child/ChannelImage';
import ChannelPostNotificationMessage from './child/ChannelPostNotificationMessage';
import ChannelPostNotificationStats from './child/ChannelPostNotificationStats';
import ChannelTitle from './child/ChannelTitle';
import CustomPressable from '../CustomPressable';
import styles from './BaseChannelItemStyles';
import {
  BaseChannelItemProps,
  BaseChannelItemTypeProps
} from '../../../types/component/AnonymousChat/BaseChannelItem.types';

const BaseChannelItem: (props: BaseChannelItemProps) => React.ReactElement = ({
  anonPostNotificationUserInfo = null,
  block = 0,
  comments = 0,
  downvote = 0,
  isCommentExists = false,
  isMe = false,
  message = '',
  name = '',
  picture = 'https://fastly.picsum.photos/id/173/400/400.jpg?hmac=TU_DMkn7FSRRwiEpVveLvmyATg-y4hNrNKT-Cs4VQ1M',
  postMaker = null,
  postNotificationMessageText = '',
  postNotificationMessageUser = null,
  postNotificationPicture = 'https://fastly.picsum.photos/id/912/400/400.jpg?hmac=Wg3Y7jTiQxHr_NpRsTrHG58kBfZQGNeH2tCl5u2Ipr0',
  time = '12:00 PM',
  type = BaseChannelItemTypeProps.ANON_PM,
  unreadCount = 0,
  upvote = 0,
  onPress = () => {
    console.log('onPress');
  }
}) => {
  return (
    <CustomPressable onPress={onPress}>
      <View style={styles.chatContainer}>
        <ChannelImage
          type={type}
          mainPicture={picture}
          postNotificationPicture={postNotificationPicture}
          anonPostNotificationUserInfo={anonPostNotificationUserInfo}
          isCommentExists={isCommentExists}
          postMaker={postMaker}
        />

        <View style={styles.chatContentContainer}>
          <ChannelTitle
            type={type}
            name={name}
            message={message}
            time={time}
            unreadCount={unreadCount}
            isMe={isMe}
          />

          {/* Post Notification Message */}
          <View style={styles.chatContentSection}>
            <ChannelPostNotificationMessage
              type={type}
              commenterName={postNotificationMessageUser}
              message={postNotificationMessageText}
            />
          </View>

          {/* Post Stats */}
          <View style={styles.chatContentSection}>
            <ChannelPostNotificationStats
              type={type}
              block={block}
              upvote={upvote}
              downvote={downvote}
              comments={comments}
            />
          </View>
        </View>
      </View>
    </CustomPressable>
  );
};

export default BaseChannelItem;
