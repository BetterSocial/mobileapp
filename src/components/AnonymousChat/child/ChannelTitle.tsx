import * as React from 'react';
import {StyleSheet, Text, View} from 'react-native';

import baseStyles from '../BaseChannelItemStyles';
import {BaseChannelItemTypeProps} from '../../../../types/component/AnonymousChat/BaseChannelItem.types';
import {colors} from '../../../utils/colors';
import {fonts} from '../../../utils/fonts';

const ChannelTitle = ({type, name, time, message, unreadCount, isMe}) => {
  const styles = StyleSheet.create({
    chatContentName: {
      fontFamily: fonts.inter[700],
      fontSize: 14.05,
      lineHeight: 22,
      alignSelf: 'center',
      flex: 1
    },
    chatContentTime: {
      fontFamily: fonts.poppins[400],
      fontSize: 14,
      lineHeight: 22,
      marginLeft: 20,
      color: colors.gray,
      alignSelf: 'flex-start'
    },
    chatContentMessage: {
      fontFamily: fonts.inter[400],
      fontSize: 14.05,
      lineHeight: 22,
      alignSelf: 'center',
      flex: 1,
      marginRight: 4
    },
    chatContentUnreadCountContainer: {
      backgroundColor: colors.bondi_blue,
      width: 20,
      height: 20,
      borderRadius: 10,
      textAlign: 'center',
      alignItems: 'center',
      justifyContent: 'center'
    },
    chatContentUnreadCount: {
      fontFamily: fonts.inter[400],
      fontSize: 12,
      lineHeight: 14.52,
      color: colors.white
    },
    chatContentUnreadCountPostNotificationContainer: {
      display: 'flex',
      position: 'absolute',
      top: 22,
      right: 20,
      alignItems: 'flex-end',
      justifyContent: 'flex-start',
      alignSelf: 'flex-start'
    },
    postNotificationMessage: {
      fontFamily: fonts.inter[400],
      alignSelf: 'flex-start'
    },
    postNotificationMessageBold: {
      fontFamily: fonts.inter[700]
    },
    anonPmMessage: {
      marginBottom: 0
    }
  });

  if (type === BaseChannelItemTypeProps.ANON_PM)
    return (
      <>
        <View style={baseStyles.chatContentSection}>
          {/* Username */}
          <Text numberOfLines={1} ellipsizeMode="tail" style={styles.chatContentName}>
            {name}
          </Text>
          {/* Time */}
          <Text style={styles.chatContentTime}>{time}</Text>
        </View>
        <View style={[baseStyles.chatContentSection, styles.anonPmMessage]}>
          {/* Message */}
          <Text numberOfLines={1} ellipsizeMode="tail" style={styles.chatContentMessage}>
            {`${isMe ? 'You: ' : ''}${message}`}
          </Text>
          {/* Unread Count */}
          {unreadCount > 0 && (
            <View style={styles.chatContentUnreadCountContainer}>
              <Text style={styles.chatContentUnreadCount}>{unreadCount}</Text>
            </View>
          )}
        </View>
      </>
    );

  const getTitle = () => {
    const isMyPost = type?.includes('MY_SIGNED_POST') || type?.includes('MY_ANON_POST');

    if (isMyPost) {
      if (message?.length === 0) return 'Your media post ðŸ“¸ ðŸ–¼';
      return 'Your post: ';
    }

    if (message?.length === 0) return `${name}'s media post ðŸ“¸ ðŸ–¼`;
    return `${name}'s post: `;
  };

  return (
    <>
      <View style={baseStyles.chatContentSection}>
        <View style={{display: 'flex', flexDirection: 'row'}}>
          <Text
            numberOfLines={2}
            ellipsizeMode="tail"
            style={[styles.chatContentName, styles.postNotificationMessage]}>
            <Text style={styles.postNotificationMessageBold}>{getTitle()}</Text>
            {message}
          </Text>
          <Text style={styles.chatContentTime}>{time}</Text>
        </View>
      </View>
      <View style={styles.chatContentUnreadCountPostNotificationContainer}>
        {unreadCount > 0 && (
          <View style={styles.chatContentUnreadCountContainer}>
            <Text style={styles.chatContentUnreadCount}>{unreadCount}</Text>
          </View>
        )}
      </View>
    </>
  );
};

export default ChannelTitle;
