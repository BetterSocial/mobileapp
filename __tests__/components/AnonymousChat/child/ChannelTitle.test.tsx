import * as React from 'react';
import {render} from '@testing-library/react-native';

import ChannelTitle from '../../../../src/components/AnonymousChat/child/ChannelTitle';
import {BaseChannelItemTypeProps} from '../../../../types/component/AnonymousChat/BaseChannelItem.types';

describe('TESTING ChatItemTargetText', () => {
  it('RENDER ANON_PM should match snapshot', () => {
    const {toJSON} = render(
      <ChannelTitle
        type={BaseChannelItemTypeProps.ANON_PM}
        name="name"
        time="time"
        message="message"
        unreadCount={0}
        isMe={false}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('RENDER ANON_PM with unreadcount should match snapshot', () => {
    const {toJSON} = render(
      <ChannelTitle
        type={BaseChannelItemTypeProps.ANON_PM}
        name="name"
        time="time"
        message="message"
        unreadCount={1}
        isMe={false}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('RENDER ANON_PM with isMe should match snapshot', () => {
    const {toJSON} = render(
      <ChannelTitle
        type={BaseChannelItemTypeProps.ANON_PM}
        name="name"
        time="time"
        message="message"
        unreadCount={0}
        isMe={true}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('RENDER MY_ANON_POST_NOTIFICATION should match snapshot', () => {
    const {toJSON} = render(
      <ChannelTitle
        type={BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION}
        name="name"
        time="time"
        message="message"
        unreadCount={0}
        isMe={false}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('RENDER MY_ANON_POST_NOTIFICATION_I_COMMENTED should match snapshot', () => {
    const {toJSON} = render(
      <ChannelTitle
        type={BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION_I_COMMENTED}
        name="name"
        time="time"
        message="message"
        unreadCount={0}
        isMe={false}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('RENDER MY_ANON_POST with no messages should match snapshot', () => {
    const {toJSON} = render(
      <ChannelTitle
        type={BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION}
        name="name"
        time="time"
        message=""
        unreadCount={0}
        isMe={false}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('RENDER other type post with messages should match snapshot', () => {
    const {toJSON} = render(
      <ChannelTitle
        type={BaseChannelItemTypeProps.ANON_POST_NOTIFICATION}
        name="name"
        time="time"
        message="message"
        unreadCount={0}
        isMe={false}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('RENDER other type post with no messages should match snapshot', () => {
    const {toJSON} = render(
      <ChannelTitle
        type={BaseChannelItemTypeProps.ANON_POST_NOTIFICATION}
        name="name"
        time="time"
        message=""
        unreadCount={0}
        isMe={false}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('RENDER other type post with unreadcount should match snapshot', () => {
    const {toJSON} = render(
      <ChannelTitle
        type={BaseChannelItemTypeProps.ANON_POST_NOTIFICATION}
        name="name"
        time="time"
        message=""
        unreadCount={1}
        isMe={false}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
