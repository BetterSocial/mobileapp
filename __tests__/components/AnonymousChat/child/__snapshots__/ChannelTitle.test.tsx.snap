// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING ChatItemTargetText RENDER ANON_PM should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "flex": 1,
            "fontFamily": "Inter-Bold",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "paddingBottom": 2,
          }
        }
      >
        name
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#828282",
            "fontFamily": "Poppins-Regular",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "marginLeft": 20,
          }
        }
      >
        time
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 0,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "alignSelf": "center",
            "flex": 1,
            "fontFamily": "Inter-Regular",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "marginRight": 4,
          }
        }
      >
        message
      </Text>
    </View>
  </View>
</View>
`;

exports[`TESTING ChatItemTargetText RENDER ANON_PM with isMe should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "flex": 1,
            "fontFamily": "Inter-Bold",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "paddingBottom": 2,
          }
        }
      >
        name
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#828282",
            "fontFamily": "Poppins-Regular",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "marginLeft": 20,
          }
        }
      >
        time
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 0,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "alignSelf": "center",
            "flex": 1,
            "fontFamily": "Inter-Regular",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "marginRight": 4,
          }
        }
      >
        You: message
      </Text>
    </View>
  </View>
</View>
`;

exports[`TESTING ChatItemTargetText RENDER ANON_PM with unreadcount should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "flex": 1,
            "fontFamily": "Inter-Bold",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "paddingBottom": 2,
          }
        }
      >
        name
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#828282",
            "fontFamily": "Poppins-Regular",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "marginLeft": 20,
          }
        }
      >
        time
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 0,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "alignSelf": "center",
            "flex": 1,
            "fontFamily": "Inter-Regular",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "marginRight": 4,
          }
        }
      >
        message
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#0391FB",
            "borderRadius": 10,
            "height": 20,
            "justifyContent": "center",
            "textAlign": "center",
            "width": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Inter-Regular",
              "fontSize": 9,
              "lineHeight": 14.52,
            }
          }
        >
          1
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`TESTING ChatItemTargetText RENDER MY_ANON_POST with no messages should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "flex": 1,
            "fontFamily": "Inter-Bold",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "paddingBottom": 2,
          },
          Object {
            "alignSelf": "flex-start",
            "fontFamily": "Inter-Regular",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Inter-Bold",
          }
        }
      >
        name's media post ðŸ“¸ ðŸ–¼
      </Text>
    </Text>
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#828282",
            "fontFamily": "Poppins-Regular",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "marginLeft": 20,
          }
        }
      >
        time
      </Text>
    </View>
  </View>
</View>
`;

exports[`TESTING ChatItemTargetText RENDER MY_ANON_POST_NOTIFICATION should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "flex": 1,
            "fontFamily": "Inter-Bold",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "paddingBottom": 2,
          },
          Object {
            "alignSelf": "flex-start",
            "fontFamily": "Inter-Regular",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Inter-Bold",
          }
        }
      >
        name's post: 
      </Text>
      message
    </Text>
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#828282",
            "fontFamily": "Poppins-Regular",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "marginLeft": 20,
          }
        }
      >
        time
      </Text>
    </View>
  </View>
</View>
`;

exports[`TESTING ChatItemTargetText RENDER MY_ANON_POST_NOTIFICATION_I_COMMENTED should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "flex": 1,
            "fontFamily": "Inter-Bold",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "paddingBottom": 2,
          },
          Object {
            "alignSelf": "flex-start",
            "fontFamily": "Inter-Regular",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Inter-Bold",
          }
        }
      >
        name's post: 
      </Text>
      message
    </Text>
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#828282",
            "fontFamily": "Poppins-Regular",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "marginLeft": 20,
          }
        }
      >
        time
      </Text>
    </View>
  </View>
</View>
`;

exports[`TESTING ChatItemTargetText RENDER other type post with messages should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "flex": 1,
            "fontFamily": "Inter-Bold",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "paddingBottom": 2,
          },
          Object {
            "alignSelf": "flex-start",
            "fontFamily": "Inter-Regular",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Inter-Bold",
          }
        }
      >
        name's post: 
      </Text>
      message
    </Text>
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#828282",
            "fontFamily": "Poppins-Regular",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "marginLeft": 20,
          }
        }
      >
        time
      </Text>
    </View>
  </View>
</View>
`;

exports[`TESTING ChatItemTargetText RENDER other type post with no messages should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "flex": 1,
            "fontFamily": "Inter-Bold",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "paddingBottom": 2,
          },
          Object {
            "alignSelf": "flex-start",
            "fontFamily": "Inter-Regular",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Inter-Bold",
          }
        }
      >
        name's media post ðŸ“¸ ðŸ–¼
      </Text>
    </Text>
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#828282",
            "fontFamily": "Poppins-Regular",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "marginLeft": 20,
          }
        }
      >
        time
      </Text>
    </View>
  </View>
</View>
`;

exports[`TESTING ChatItemTargetText RENDER other type post with unreadcount should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "flex": 1,
            "fontFamily": "Inter-Bold",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "paddingBottom": 2,
          },
          Object {
            "alignSelf": "flex-start",
            "fontFamily": "Inter-Regular",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Inter-Bold",
          }
        }
      >
        name's media post ðŸ“¸ ðŸ–¼
      </Text>
    </Text>
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#828282",
            "fontFamily": "Poppins-Regular",
            "fontSize": 12.600000000000001,
            "lineHeight": 22,
            "marginLeft": 20,
          }
        }
      >
        time
      </Text>
      <View
        style={
          Object {
            "alignSelf": "flex-end",
            "position": "absolute",
            "top": 24,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#0391FB",
              "borderRadius": 10,
              "height": 20,
              "justifyContent": "center",
              "textAlign": "center",
              "width": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Inter-Regular",
                "fontSize": 9,
                "lineHeight": 14.52,
              }
            }
          >
            1
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
